spec-examples/src/ntru_prime/ntru_prime.rs

```diff
@@ -0,0 +1,114 @@
+// Import hacspec and all needed definitions.
+use hacspec::prelude::*;
+
+
+/// Struct to decide NtruVersion
+pub struct NtruVersion{
+    pub p:i128,
+    pub q:i128,
+    pub w:i128,
+}
+#[macro_export]
+macro_rules! ntru_v {
+    ($t:expr)  => {
+        {
+        // TODO needs to be removed, just for demonstration
+        if $t == -1{
+            NtruVersion{p:191,q:397,w:68}
+        }
+        else if $t == 0{
+            NtruVersion{p:761,q:4591,w:286}
+        } else if $t == 1{
+            NtruVersion{p:653,q:4621,w:288}
+        } else {
+            NtruVersion{p:857,q:5167,w:322}
+        }
+
+        }
+    };
+
+}
+
+
+///This function creates a random polynom with w many -1 or 1 and with the highes degree of h_deg.
+fn create_rand_poly(w: i128, h_deg: usize) -> Vec<(usize, i128)> {
+    let mut counter = 0;
+    let mut pos;
+    let mut polynom: Vec<(usize, i128)> = Vec::new();
+    while w != counter {
+        pos = rand::thread_rng().gen_range(0, h_deg);
+        let c_val = rand::thread_rng().gen_range(0, 2);
+        if polynom.contains(&(pos, -1)) || polynom.contains(&(pos, 1)) {
+            continue;
+        }
+        // if c_val = 0 -> polynom[pos] = -1, else 1
+        polynom.push((pos, c_val * 2 - 1));
+        counter = counter + 1;
+    }
+
+    return polynom;
+}
+///Creates a random polynom with the inverse, if there is no inverse this function return []
+//TODO for tests pub
+
+pub fn create_invertable_poly(n: NtruVersion, modulus: i128) -> ([(usize, u128); 288], Vec<u128>) /*-> Result<(),Box<dyn Error>>*/
```

Comment:
Again, no `Vec`.

Reply:
If `[(usize, u128); 288]` is something you need, you should create an `array!` for it. Plain Rust arrays aren't allowed either.

Reply:
This function should return a `Result`.